test/main/store/migrations/index.test.js

```diff
@@ -528,3 +528,279 @@ describe('migration 20', () => {
+    state.main.networks.ethereum[5].connection.primary = { 
+      on: false, 
+      current: 'prylabs', 
+      status: 'disconnected', 
+      connected: false, 
+      type: '', 
+      network: '', 
+      custom: '' 
+    }
+    state.main.networks.ethereum[5].connection.secondary = { 
+      on: false, 
+      current: 'infura', 
+      status: 'loading', 
+      connected: false, 
+      type: '', 
+      network: '', 
+      custom: '' 
+    }
+
+    const updatedState = migrations.apply(state, 21)
+    const goerli = updatedState.main.networks.ethereum[5]
+
+    expect(goerli.connection).toMatchObject({
+      primary: { on: false, current: 'custom', status: 'loading', connected: false, type: '', network: '', custom: '' },
+      secondary: { on: false, current: 'infura', status: 'loading', connected: false, type: '', network: '', custom: '' }
+    })
+    expect(goerli.on).toBe(false)
+  })
+
+  it('turns off goerli if the secondary connection was reset whilst the primary connection is inactive', () => {
+    state.main.networks.ethereum[5].connection.primary = { 
+      on: false, 
+      current: 'infura', 
+      status: 'loading', 
+      connected: false, 
+      type: '', 
+      network: '', 
+      custom: '' 
+    }
+    state.main.networks.ethereum[5].connection.secondary = { 
+      on: false, 
+      current: 'prylabs', 
+      status: 'disconnected', 
+      connected: false, 
+      type: '', 
+      network: '', 
+      custom: '' 
+    }
+
+    const updatedState = migrations.apply(state, 21)
+    const goerli = updatedState.main.networks.ethereum[5]
+
+    expect(goerli.connection).toMatchObject({
+      primary: { on: false, current: 'infura', status: 'loading', connected: false, type: '', network: '', custom: '' },
+      secondary: { on: false, current: 'custom', status: 'loading', connected: false, type: '', network: '', custom: '' }
+    })
+    expect(goerli.on).toBe(false)
+  })
+
+  it('turns off goerli if both connections were reset', () => {
+    state.main.networks.ethereum[5].connection.primary = { 
+      on: true, 
+      current: 'mudit', 
+      status: 'connected', 
+      connected: true, 
+      type: '', 
+      network: '', 
+      custom: '' 
+    }
+    state.main.networks.ethereum[5].connection.secondary = { 
+      on: false, 
+      current: 'prylabs', 
+      status: 'disconnected', 
+      connected: false, 
+      type: '', 
+      network: '', 
+      custom: '' 
+    }
+
+    const updatedState = migrations.apply(state, 21)
+    const goerli = updatedState.main.networks.ethereum[5]
+
+    expect(goerli.connection).toMatchObject({
+      primary: { on: false, current: 'custom', status: 'loading', connected: false, type: '', network: '', custom: '' },
+      secondary: { on: false, current: 'custom', status: 'loading', connected: false, type: '', network: '', custom: '' }
+    })
+    expect(goerli.on).toBe(false)
```

Comment:
for this test we only need this expectation, the previous one is tested elsewhere and isn't the focus of this particular test

Reply:
same goes for the other 2 tests in this category